2to3.patch

```diff
@@ -1,270 +1,57 @@
 --- dbbackup/management/commands/dbrestore.py	(original)
 +++ dbbackup/management/commands/dbrestore.py	(refactored)
-@@ -39,7 +39,7 @@
-             if options.get('list'):
-                 return self.list_backups()
-             self.restore_backup()
--        except StorageError, err:
-+        except StorageError as err:
-             raise CommandError(err)
- 
-     def _get_database(self, options):
-@@ -50,22 +50,22 @@
-                 errmsg = "Because this project contains more than one database, you"
-                 errmsg += " must specify the --database option."
-                 raise CommandError(errmsg)
--            database_key = settings.DATABASES.keys()[0]
-+            database_key = list(settings.DATABASES.keys())[0]
-         return settings.DATABASES[database_key]
- 
-     def restore_backup(self):
-         """ Restore the specified database. """
--        print "Restoring backup for database: %s" % self.database['NAME']
-+        print("Restoring backup for database: %s" % self.database['NAME'])
-         # Fetch the latest backup if filepath not specified
+@@ -76,7 +76,7 @@
          if not self.filepath:
--            print "  Finding latest backup"
-+            print("  Finding latest backup")
+             print("  Finding latest backup")
              filepaths = self.storage.list_directory()
-             filepaths = self.dbcommands.filter_filepaths(filepaths, self.servername)
+-            filepaths = list(filter(lambda f: f.endswith('.' + self.backup_extension), filepaths))
++            filepaths = list([f for f in filepaths if f.endswith('.' + self.backup_extension)])
              if not filepaths:
-                 raise CommandError("No backup files found in: /%s" % self.storage.backup_dir())
+                 raise CommandError("No backup files found in: /%s" % self.storage.backup_dir)
              self.filepath = filepaths[-1]
-         # Restore the specified filepath backup
--        print "  Restoring: %s" % self.filepath
-+        print("  Restoring: %s" % self.filepath)
-         input_filename = self.filepath
-         inputfile = self.storage.read_file(input_filename)
-         if self.get_extension(input_filename) == '.gpg':
-@@ -77,7 +77,7 @@
-             uncompressed_file = self.uncompress_file(inputfile)
+@@ -94,7 +94,7 @@
              inputfile.close()
              inputfile = uncompressed_file
--        print "  Restore tempfile created: %s" % utils.handle_size(inputfile)
-+        print("  Restore tempfile created: %s" % utils.handle_size(inputfile))
-         self.dbcommands.run_restore_commands(inputfile)
- 
-     def get_extension(self, filename):
-@@ -99,8 +99,8 @@
-         """ Unencrypt this file using gpg. The input and the output are filelike objects. """
+         print("  Restore tempfile created: %s" % utils.handle_size(inputfile))
+-        cont = input("Are you sure you want to continue? [Y/n]")
++        cont = eval(input("Are you sure you want to continue? [Y/n]"))
```

Comment:
Wondering... What's the eval() expression's purpose? Would seem like a bad idea for user input...